students/Thai_H/lesson6/mail_room_part4.py

```diff
@@ -0,0 +1,222 @@
+        #test_create_report()
+
+        # Add amount to existing donor
+        #list_of_donors[donor_name].append(donation_amt)
+        add_donation(donor_name, donation_amt,'E')
+
+        #--- instead of using this block ------------------------------------
+        #for each_donor in list_of_donors:
+        #    if donor_name == each_donor['name']:
+        #        each_donor['donation'].append(float(donation_amt))
+        #-----------------------------------------------------------------
+
+        #we will use this python list comprehension
+        #[each_donor['donation'].append(float(donation_amt)) for each_donor in list_of_donors if donor_name == each_donor['name']]
+        #test_create_report()
+
+    # New donor
+    else:
+        #list_of_donors[donor_name] = []
+        #list_of_donors[donor_name].append(donation_amt)
+        add_donation(donor_name, donation_amt,'N')
+    #send thank you letter to the the donor name user just entered
+    send_thank_you_letter(donor_name, donation_amt)
+
+
+#----------------------------------------------------------------------------
+def create_thank_you_files():
+    for donor_name in list_of_donors.keys():
+        total_donation = sum(list_of_donors[donor_name]) #sum(each_donor['donation'])
+        thks_letter = open('{name}.txt'.format(name = donor_name), 'w')
+        thks_letter.write('Dear {name}, \n'
+                          'Thank you for your generosity to our Foundation in the total amount of ${amt}'.format(name = donor_name,
+                                                                                                                 amt = total_donation))
+        thks_letter.close()
+
+#----------------------------------------------------------------------------
+def thank_you_menu_selection():
+    user_selection = input('\n\nTHANK-YOU MENU\nChoose from one of options below:\n\
+        L) List current donors\n\
+        T) Send Thank-you letter (or Add donor and Send one)\n\
+        F) Create/Write Thank-you files\n\
+        M) Back to MAIN MENU\n\
+        Q) Quit the program\n\
+        Please type L, T, F, M, or Q: ')
+    return user_selection
+
+
+#-------------------------------------------------------------------------
+def thank_you_letter():
+    while True:
+        # use of dictionary where 'key' is the menu choice and 'value' is function to call
+        thank_you_menu_function = {
+        'L': create_report,
+        'T': send_thank_you_add_new_donation,
+        'F': create_thank_you_files,
+        'M': main,
+        #'Z': test_create_report,
+        'Q': quit_program
+    }
+        thank_you_selection = thank_you_menu_selection().upper()
+        thank_you_menu_function.get(thank_you_selection, invalid_thank_you_menu)()
+        #invalid_menu is to handle non-of-above choices
+                                                                #
+
+#----------------------------------------
+def sum_donation(d_amt):
+    return sum(d_amt)
```

Comment:
from the "learning how to unit test" point of view - yes this is a good way to practice that, however, this is only testing built-in `sum` function which we already know works.

You should concentrate on testing your code rather than internals - and no need to wrap built-in function with your own functions.